components/eamxx/src/diagnostics/tests/field_at_height_tests.cpp

```diff
@@ -133,29 +132,39 @@ TEST_CASE("field_at_height")
   //
   // To simplify the surface contribution we set the surface height to equal
   // the local column index.
-  const float z_top = 10.0;
+  const int z_top = 1000;
+  const Real surf_slope = z_top/10.0/ncols;
   const auto& zint_v   = z_int.get_view<Real**,Host>();
   const auto& zmid_v   = z_mid.get_view<Real**,Host>();
   const auto& zsurf_v  = z_surf.get_view<Real*,Host>();
   const auto& geoint_v = geo_int.get_view<Real**,Host>();
   const auto& geomid_v = geo_mid.get_view<Real**,Host>();
+  int         min_col_thickness = z_top;
+  int         max_surf = 0;
   for (int ii=0; ii<ncols; ++ii) {
-    zsurf_v(ii) = ii;
-    const float dz = (z_top - zsurf_v(ii))/nlevs;
+    zsurf_v(ii) = ii*surf_slope;
+    max_surf = zsurf_v(ii) > max_surf ? zsurf_v(ii) : max_surf;
+    const Real col_thickness = z_top - zsurf_v(ii);
+    min_col_thickness = min_col_thickness < col_thickness ? col_thickness : min_col_thickness;
+    const Real dz = (z_top - zsurf_v(ii))/nlevs;
```

Comment:
I believe your approach works, since `ztop-zsurf` is larger than `nlevs`. However, I would like to make sure this won't change in the future (e.g., someone changes this test 1y from now, making ztop=10).

Can you add something like this?

```suggestion
REQUIRE (dz>1);
```

If I understand correctly, given your choices, right now we have
- slope = 100/ncols=33.3
- max(zsurf) = (ncols-1)*slope=66.6
- min(dz) = (1000-max(zsurf))/nlevs = 933.3/10 = 93.3

So we're good. But better make sure this stays like this.

Reply:
Since committing now would abort the AT run, you can do this change later, and merge without another round of testing. I'm confident that the data we have would make the `REQUIRE(dz>1)` check pass.

Reply:
You could maybe run the test locally on your machine, to make sure our math is correct, and dz>1 always (meaning that check will pass, and we're not merging something that will cause fails)